### DISCUSSION TITLE
New HV battery: Reset charging data?

### DISCUSSION THREAD

**Discussion by cg089** [teslamate-org/teslamate#3365] [General]
Unfortunately, my Model 3 gets a new HV battery.

What is the best practise with all my data in the app? I would like to track the new battery from scratch (AC:DC ratio etc). Starting completely new will lose my driving data of 2 years.

Any ideas?

**Comment by cwanja** [comment]
How important is the old HV battery charging data? Do you want the old charging data? Or okay departing with it (archive)? 

Could backup the entire database and then wipe the charges table. But you would not be able to see the old charging data without restoring and that could get messy.

**Reply by cg089** [reply]
I would need the old charging data only for tax (you have to prove how many kwh you charged at home), other than that it could be fully deleted.

**Reply by cwanja** [reply]
In that event, I would take a backup and wipe the charges table. You can take a backup at that when you prepare to file taxes, restore today’s backup temporarily next year when you have to file taxes (assumptions made) and then restore the known working database all without losing data. 

Open to others thoughts however too. 

Alternative would be to keep it all the same, record the date the HV is replaced and then try to backup the data and delete all data prior to the HV replacement date in the future when you file your taxes.

**Reply by cg089** [reply]
Thanks :)

I thought about adding it as a new car, however I am not sure how that would work since it is the same VIN and there still is only 1 car in my account.

**Reply by cwanja** [reply]
That is not a bad idea, although I am not sure you can update the ID of the vehicles table - believe it is the primary key. You could try it and see what happens. You would need to make every charge session associated with vehicle ID 2 as well. Then start TeslaMate service and see what is added back. As you said, given the same VIN and vehicle ID, TeslaMate get confused.

**Comment by sparvata** [comment]
or you can run two instances of teslamate on the same host. To prevent one instance from updating the data you can provide some incorrect auth key so that any updates will fail. Refer to https://github.com/jheredianet/Teslamate-CustomGrafanaDashboards/discussions/56#discussioncomment-7095047 for more details on the second instance. 
Note that you need to back up the data and populate into the second database before resetting the data in the first instance.
